### ISSUE TITLE
Does Pyarrow support Parquet 2.0

### ISSUE THREAD

**Issue by dmgcodevil** [apache/arrow#12597]
I'm using pyarrow to read parquet files, however, it fails to read some files. 

Error:

```
  table = self._dataset.to_table(
  File "pyarrow\_dataset.pyx", line 323, in pyarrow._dataset.Dataset.to_table
  File "pyarrow\_dataset.pyx", line 2311, in pyarrow._dataset.Scanner.to_table
  File "pyarrow\error.pxi", line 143, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow\error.pxi", line 114, in pyarrow.lib.check_status
OSError: Malformed levels. min: 60 max: 60 out of range.  Max Level: 1
```

Based on my research the file that cannot be read uses a new 2.0 encoding

Column meta from the file that pyarrow  reads successfully

`id:             BINARY GZIP DO:0 FPO:630 SZ:101/63/0.62 VC:30 ENC:PLAIN_DICTIONARY,BIT_PACKED,RLE ST:[min: , max: D, num_nulls: 0]`

Column meta from the file that pyarrow  fails to read

`id:             BINARY GZIP DO:0 FPO:593 SZ:90/52/0.58 VC:30 ENC:RLE_DICTIONARY ST:[min: , max: D, num_nulls: 0]`

Based on Parquet spec, `RLE_DICTIONARY` is a new encoding format:

> Using the PLAIN_DICTIONARY enum value is deprecated in the Parquet 2.0 specification. Prefer using RLE_DICTIONARY in a data page and PLAIN in a dictionary page for Parquet 2.0+ files.

Does Pyarrow support Parquet 2.0 format ?

**Comment by jorisvandenbossche** [comment]
Yes, Arrow does generally support Parquet version 2.x (although not every feature), and specifically supports reading RLE_DICTIONARY encoding. See the overview of supported parquet features here: https://arrow.apache.org/docs/cpp/parquet.html#encodings

Did you write the file yourself, or do you know how it was written? (with which software?) 
Would you be able to share the file that reproduces the issue? (or a file with some dummy data, that if written in the same way also reproduces the issue)

**Comment by dmgcodevil** [comment]
@jorisvandenbossche , the files that Pyarrow successfully reads written by Spark/Iceberg data source and Iceberg's [ParquetWriter].(https://github.com/apache/iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java)

The files that Pyarrow fails to read are written via Trino Iceberg catalog (connector). In theory, both Trino and Spark should use Iceberg ParquetWriter which internally uses Hadoop Parquet Writer. 

What I've found is that some columns are encoded in DELTA_BYTE_ARRAY. Does Pyarrow support this encoding? I know that Fastparquet does not. I also found this [ticket](https://issues.apache.org/jira/browse/ARROW-6057?src=confmacro), is it still relevant?

**Comment by emkornfield** [comment]
DELTA_BYTE_ARRAY has been implemented as of [pyarrow 7.0](https://issues.apache.org/jira/browse/PARQUET-492?jql=text%20~%20%22DELTA_BYTE_ARRAY%22).

It [does not handle](https://issues.apache.org/jira/browse/ARROW-14089?jql=text%20~%20%22DELTA_BYTE_ARRAY%22) a bug for older writers of this encoding.

**Comment by dmgcodevil** [comment]
@emkornfield Ok, I'll keep investigating the issue. Maybe you could give me some directions? One thing I know for sure: "bad" files are 20% smaller than "good" files, although they have identical content.

**Comment by emkornfield** [comment]
Looking more closely at the error something else is going on.  If you can produce a minimal repro and open up a JIRA we can try to investigate further

**Comment by pitrou** [comment]
I'm closing since there has been no further answer. @dmgcodevil feel free to re-open or open a new issue if you want to bring further information.
